jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridView.java

```diff
@@ -377,6 +390,64 @@ public T getValue(T object) {
         return actionsColMeta;
     }
 
+    public List<ColumnMeta<T>> renameProcessVariables(ListTable<T> extendedPagedTable, List<ColumnMeta<T>> columnMetas) {
+        List<GridColumnPreference> columnPreferenceList = extendedPagedTable.getGridPreferencesStore().getColumnPreferences();
+
+        return (List<ColumnMeta<T>>) columnPreferenceList.stream().filter(colPref -> !isColumnAdded(columnMetas, colPref.getName())).map(colPref -> {
+            String columnName = colPref.getName();
+            if (existsColumnWithSameName(colPref, columnMetas)) {
+                columnName = "Var_" + columnName;
+            }
+            Column genericColumn = initGenericColumn(colPref.getName());
+            genericColumn.setSortable(false);
+            return new ColumnMeta<T>(genericColumn,
+                                     columnName,
+                                     true,
+                                     true);
+        }).collect(Collectors.toList());
+    }
+
+    public void addDomainSpecifColumns(ExtendedPagedTable<T> extendedPagedTable,
+                                       Set<String> columns) {
+        extendedPagedTable.storeColumnToPreferences();
+
+        List<ColumnMeta<T>> columnMetas = extendedPagedTable.getColumnMetaList().stream()
+                .filter(cm -> cm.isExtraColumn())
+                .collect(Collectors.toList());
+
+        columnMetas.forEach(colMet -> {
+            if (!columns.contains(colMet.getCaption())) {
+                extendedPagedTable.removeColumnMeta(colMet);
+            } else {
+                columns.remove(colMet.getCaption());
+            }
+        });
+
+        List<ColumnMeta<T>> columnMetaList = (List<ColumnMeta<T>>)extendedPagedTable.getColumnMetaList().stream()
+                .filter(cm -> !cm.isExtraColumn() && columns.contains(cm.getCaption()))
+                .map(colMet -> {
+                    Column genericColumn = initGenericColumn(colMet.getCaption());
+                    genericColumn.setSortable(false);
+                    return new ColumnMeta<T>(genericColumn,
+                                             "Var_" + colMet.getCaption(),
+                                             true,
+                                             true);
+                }).collect(Collectors.toList());
+
+
+        columns.stream().filter(column -> (columnMetaList.stream().filter(columnMeta -> column.contains(columnMeta.getColumn().getDataStoreName())).count() == 0))
+                .forEach(column -> {
+                    Column genericColumn = initGenericColumn(column);
```

Comment:
@bxf12315 this is very similar as on line 429. Could be extracted to helper method. That would also improve readability.